ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateDiff.java

```diff
@@ -64,121 +66,46 @@
+    Timestamp ts2 = getTimestampValue(arguments, 1, tsConverters);
 
-  @Nullable
-  private Date convertToDate(PrimitiveCategory inputType, Converter converter, DeferredObject argument)
-    throws HiveException {
-    assert(converter != null);
-    assert(argument != null);
-    if (argument.get() == null) {
-      return null;
-    }
-    switch (inputType) {
-    case STRING:
-    case VARCHAR:
-    case CHAR: {
-      String dateString = converter.convert(argument.get()).toString();
-      Date date = DateParser.parseDate(dateString);
-      if (date != null) {
-        return date;
-      }
-      Timestamp ts = PrimitiveObjectInspectorUtils.getTimestampFromString(dateString);
-      if (ts != null) {
-        return Date.ofEpochMilli(ts.toEpochMilli());
-      }
+    if (ts1 == null || ts2 == null) {
       return null;
     }
-    case TIMESTAMP:
-      Timestamp ts = ((TimestampWritableV2) converter.convert(argument.get()))
-        .getTimestamp();
-      return Date.ofEpochMilli(ts.toEpochMilli());
-    case DATE:
-      DateWritableV2 dw = (DateWritableV2) converter.convert(argument.get());
-      return dw.get();
-    case TIMESTAMPLOCALTZ:
-      TimestampTZ tsz = ((TimestampLocalTZWritable) converter.convert(argument.get()))
-          .getTimestampTZ();
-      return Date.ofEpochMilli(tsz.getEpochSecond() * 1000l);
-    default:
-      throw new UDFArgumentException(
-        "TO_DATE() only takes STRING/TIMESTAMP/TIMESTAMPLOCALTZ types, got " + inputType);
-    }
-  }
 
-  private Converter checkArguments(ObjectInspector[] arguments, int i) throws UDFArgumentException {
-    if (arguments[i].getCategory() != ObjectInspector.Category.PRIMITIVE) {
-      throw new UDFArgumentTypeException(0,
-        "Only primitive type arguments are accepted but "
-        + arguments[i].getTypeName() + " is passed. as first arguments");
-    }
-    final PrimitiveCategory inputType =
-        ((PrimitiveObjectInspector) arguments[i]).getPrimitiveCategory();
-    switch (inputType) {
-    case STRING:
-    case VARCHAR:
-    case CHAR:
-      return ObjectInspectorConverters.getConverter(arguments[i],
-        PrimitiveObjectInspectorFactory.writableStringObjectInspector);
-    case TIMESTAMP:
-      return new TimestampConverter((PrimitiveObjectInspector) arguments[i],
-        PrimitiveObjectInspectorFactory.writableTimestampObjectInspector);
-    case TIMESTAMPLOCALTZ:
-      return new PrimitiveObjectInspectorConverter.TimestampLocalTZConverter(
-          (PrimitiveObjectInspector) arguments[i],
-          PrimitiveObjectInspectorFactory.writableTimestampTZObjectInspector
-      );
-    case DATE:
-      return ObjectInspectorConverters.getConverter(arguments[i],
-        PrimitiveObjectInspectorFactory.writableDateObjectInspector);
-    default:
-      throw new UDFArgumentException(
-          " DATEDIFF() only takes STRING/TIMESTAMP/DATEWRITABLE/TIMESTAMPLOCALTZ types as " + (i + 1)
-              + "-th argument, got " + inputType);
```

Comment:
Same comment here, do we have test coverage for this?

Reply:
Yes we have couple of qtest which checks both positive and negative. Include command like "DESCRIBE FUNCTION EXTENDED datediff";